src/irmin-pack/inode.ml

```diff
@@ -992,6 +1030,99 @@ struct
       check_lower t
 
     let is_tree t = match t.v with Tree _ -> true | Values _ -> false
+
+    type nonrec proof = (hash, step, value) Irmin.Private.Node.Proof.t
+    [@@deriving irmin]
+
+    module Proof = struct
+      let rec proof_of_concrete h : Concrete.t -> proof = function
+        | Blinded -> Blinded (Lazy.force h)
+        | Values vs -> Values (List.map Concrete.of_entry vs)
+        | Tree tr ->
+            let proofs =
+              List.fold_left
+                (fun acc (e : _ Concrete.pointer) ->
+                  let p = proof_of_concrete (lazy e.pointer) e.tree in
+                  let e =
+                    match p with
+                    | Inode { proofs = [ (index, proof) ]; _ } ->
+                        (e.index :: index, proof)
```

Comment:
Is this extender pattern occurring at all in your tests? It think that the probability it occurs in our design is very close to zero.

Reply:
yes I'm testing the extender - I think they are used in ethereum to protect against attacks (where people will try to build a very unbalanced tree by using a lot of computing power) and then using this to creat huge proofs.